### PR TITLE
Climb

### PR DESCRIPTION
**Pull request by Ethan14123**
It pick itself up from the ground and try not to fall to the ground and try's to spits into the slot.

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -69,5 +69,18 @@ public static final class Swerve {
...
+  public static final class Climb{
+    public static final double MAX_UP_VOLTAGE = 0.4; 
+    public static final double MAX_DOWN_VOLTAGE= -0.4; 
+    public static final double EXTEND_MOTOR_SETPIONT = 0.9; 
+    public static final double RETRACT_MOTOR_SETPOINT = 0.9;
+    public static final int RCLIMB_MOTOR_ID = 14;
+    public static final int LCLIMB_MOTOR_ID = 15;
+    public static final double CLIMB_kP = .1;
+    public static final double CLIMB_kI = .1;
+    public static final double CLIMB_kD = .1;
```
Comment (LucasF6) [inline src/main/java/frc/robot/Constants.java]: The P, I, and D gains should be left as zero for now.

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -6,16 +6,22 @@
...
 import frc.robot.commands.DefaultDrive;
 import frc.robot.commands.FieldDrive;
+import frc.robot.subsystems.ClimbSubsystem;
 import frc.robot.subsystems.SwerveSubsystem;
 
 import static frc.robot.Constants.Swerve.SQUARED_INPUTS;
 
 public class RobotContainer {
-  private CommandXboxController m_controller = new CommandXboxController(1);
+  private CommandJoystick m_Joystick = new CommandJoystick(0);
```
Comment (LucasF6) [inline src/main/java/frc/robot/RobotContainer.java]: This doesn't follow the naming convention. It should be m_joystick

File: src/main/java/frc/robot/subsystems/ClimbSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    double m_leftCurrentHeight = m_leftClimbEncoder.getPosition();
+    
+    double m_rightMotorOutput = m_rightPIDController.calculate(m_rightCurrentHeight);
+    double m_leftMotorOutput = m_leftPIDController.calculate(m_leftCurrentHeight);
+   
+  
+    m_rightClimbMotor.set(MathUtil.clamp(m_rightMotorOutput, MAX_DOWN_VOLTAGE,MAX_UP_VOLTAGE )); 
+    m_leftClimbMotor.set(MathUtil.clamp(m_leftMotorOutput, MAX_DOWN_VOLTAGE, MAX_UP_VOLTAGE)); 
+    m_leftClimbEncoder.setPosition(m_leftMotorOutput); 
+    m_rightClimbEncoder.setPosition(m_rightMotorOutput);
```
Comment (LucasF6) [inline src/main/java/frc/robot/subsystems/ClimbSubsystem.java]: The only time you should set the position of the encoder is during initialization. These two lines should not be here.

File: src/main/java/frc/robot/subsystems/ClimbSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+  public void retract() {   
+    m_rightPIDController.setSetpoint(RETRACT_MOTOR_SETPOINT);
+    m_leftPIDController.setSetpoint(RETRACT_MOTOR_SETPOINT);
+  }
+ /*
+  * Makes a command that extends the climb
+  */
+public Command getExtend(){
+  return runOnce(this::extend);
```
Comment (LucasF6) [inline src/main/java/frc/robot/subsystems/ClimbSubsystem.java:64]: To name the command _getExtend_ doesn't exactly make sense. The name of this method should be getExtendCommand

**Review by LucasF6 — CHANGES_REQUESTED** [review]
There needs to be a change in the periodic method of the climb subsystem.

File: src/main/java/frc/robot/Constants.java (hunk 1 above)
Comment (Ethan14123) [reply src/main/java/frc/robot/Constants.java]: solved

**Review by LucasF6 — APPROVED** [review]

**Pull request merged**